homeassistant/components/light/rest.py

```diff
@@ -0,0 +1,440 @@
+DEFAULT_IS_ON_TEMPLATE = Template('{{value_json.is_on}}')
+DEFAULT_BRIGHTNESS_TEMPLATE = Template('{{value_json.brightness}}')
+DEFAULT_COLOR_TEMP_TEMPLATE = Template('{{value_json.color_temp}}')
+DEFAULT_EFFECT_TEMPLATE = Template('{{value_json.effect}}')
+DEFAULT_RGB_COLOR_TEMPLATE = Template('{{value_json.rgb_color}}')
+DEFAULT_NAME_TEMPLATE = Template('{{value_json.name}}')
+DEFAULT_SUPPORTED_FEATURES_TEMPLATE = Template(
+    '{{value_json.supported_features}}')
+DEFAULT_EFFECT_LIST_TEMPLATE = Template('{{value_json.effect_list}}')
+DEFAULT_SUPPORTED_FEATURES = ['brightness']
+DEFAULT_NAME = 'REST Light'
+DEFAULT_TIMEOUT = 10
+
+SUPPORT_REST_METHODS = ['post', 'put']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL_ON): cv.url,
+    vol.Required(CONF_URL_OFF): cv.url,
+    vol.Required(CONF_URL_STATE): cv.url,
+    vol.Optional(CONF_BODY_OFF, default=DEFAULT_BODY_OFF): cv.template,
+    vol.Optional(CONF_BODY_ON, default=DEFAULT_BODY_ON): cv.template,
+    vol.Optional(CONF_BODY_BRIGHTNESS,
+                 default=DEFAULT_BODY_BRIGHTNESS): cv.template,
+    vol.Optional(CONF_BODY_COLOR_TEMP,
+                 default=DEFAULT_BODY_COLOR_TEMP): cv.template,
+    vol.Optional(CONF_BODY_EFFECT, default=DEFAULT_BODY_EFFECT): cv.template,
+    vol.Optional(CONF_BODY_RGB_COLOR,
+                 default=DEFAULT_BODY_RGB_COLOR): cv.template,
+    vol.Optional(CONF_BODY_TRANSITION,
+                 default=DEFAULT_BODY_TRANSITION): cv.template,
+    vol.Optional(CONF_IS_ON_TEMPLATE,
+                 default=DEFAULT_IS_ON_TEMPLATE): cv.template,
+    vol.Optional(CONF_BRIGHTNESS_TEMPLATE,
+                 default=DEFAULT_BRIGHTNESS_TEMPLATE): cv.template,
+    vol.Optional(CONF_COLOR_TEMP_TEMPLATE,
+                 default=DEFAULT_COLOR_TEMP_TEMPLATE): cv.template,
+    vol.Optional(CONF_EFFECT_TEMPLATE,
+                 default=DEFAULT_EFFECT_TEMPLATE): cv.template,
+    vol.Optional(CONF_RGB_COLOR_TEMPLATE,
+                 default=DEFAULT_RGB_COLOR_TEMPLATE): cv.template,
+    vol.Optional(CONF_NAME_TEMPLATE,
+                 default=DEFAULT_NAME_TEMPLATE): cv.template,
+    vol.Optional(CONF_SUPPORTED_FEATURES_TEMPLATE,
+                 default=DEFAULT_SUPPORTED_FEATURES_TEMPLATE): cv.template,
+    vol.Optional(CONF_EFFECT_LIST_TEMPLATE,
+                 default=DEFAULT_EFFECT_LIST_TEMPLATE): cv.template,
+    vol.Optional(CONF_METHOD, default=DEFAULT_METHOD):
+        vol.All(vol.Lower, vol.In(SUPPORT_REST_METHODS)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_SUPPORTED_FEATURES,
+                 default=DEFAULT_SUPPORTED_FEATURES): vol.Coerce(tuple),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    vol.Inclusive(CONF_USERNAME, 'authentication'): cv.string,
+    vol.Inclusive(CONF_PASSWORD, 'authentication'): cv.string,
+})
+
+
+# pylint: disable=unused-argument
+@asyncio.coroutine
```

Comment:
We've migrated to `async`/`await` syntax. Please drop this decorator and prefix async methods with `async` like `async def async_setup_platform`. Replace any `yield from` with `await`.